### PR TITLE
More Pandaria Stuff

### PR DESCRIPTION
**Pull request by Shoxxo**
- Adding some Pandaria Zones
- Adding some Pandaria Bosses
- Adding some Pandaria Rares
- Adding OnAuraUpdate += AuraUpdateFn to Core
- Passed some scripts to the Core
- Deleted some old Trash folders and non used shit.

### CODE REVIEW

File: src/server/game/Spells/Auras/SpellAuras.cpp [hunk 1]
```diff
@@ -744,6 +744,72 @@ void Aura::UpdateOwner(uint32 diff, WorldObject* owner)
     _DeleteRemovedApplications();
 }
```
Comment (Bootz) [inline src/server/game/Spells/Auras/SpellAuras.cpp]: issues | reactions
------------ | -------------
!Moar Tabs  uhhNo... | them01 
the magic is in the numbers, would using a switch with case(s)?? `@lines: 759-763` | them01
Theres something confusing gone on here??!! @line: 2085 `for (; effItr != effEndItr; ++effItr)` |

Comment (Shoxxo) [reply src/server/game/Spells/Auras/SpellAuras.cpp]: Some idiot removed my push rights normaly i will merge it but okay than close it .. it is what it is. i asked 100 times for editor configs noone sends me shit happens. live with dirty struct in folders etc.

Comment (Shoxxo) [reply src/server/game/Spells/Auras/SpellAuras.cpp]: @Bootz https://www.projectskyfire.org/index.php?/topic/10429-dev-rights/&tab=comments#comment-25777

File: src/server/game/Spells/Auras/SpellAuras.cpp [hunk 2]
```diff
@@ -744,6 +744,72 @@ void Aura::UpdateOwner(uint32 diff, WorldObject* owner)
...
 
+void Aura::Update(uint32 diff, Unit* caster)
```
Comment (AriDEV) [inline src/server/game/Spells/Auras/SpellAuras.cpp]: this function allready exists on line 813

Comment (AriDEV) [reply src/server/game/Spells/Auras/SpellAuras.cpp]: instead of tabs use 4 spaces pls setup your configs in msvc

Comment (AriDEV) [reply src/server/game/Spells/Auras/SpellAuras.cpp]: most of the scripts need to be rewriten and linked with db most of them never worked

File: src/server/game/Spells/Auras/SpellAuras.cpp (hunk 1 above)
Comment (Bootz) [reply src/server/game/Spells/Auras/SpellAuras.cpp]: yeah ive no part of the dev-right situation, /shrugs
but as for the cut-n-paste whitespace hell and tabs....
```Java
you could just embrace the Style through all your editors, `indent +2, tabs -> indent +4, spaces`.
    would save on the headache and save ya from backtracking to cleanup. 
```

File: sql/updates/world/creature_template_2018_03_09_00.sql [hunk 3]
```diff
@@ -24,3 +39,7 @@ DELETE FROM `creature_text` WHERE `entry`=72057 AND `groupid`=5 AND `id`=0;
 INSERT INTO `creature_text` (`entry`, `groupid`, `id`, `text`, `type`, `language`, `probability`, `emote`, `duration`, `sound`, `comment`) VALUES (72057, 5, 0, 'You will feel but a fraction of my agony.', 14, 0, 100, 0, 0, 38980, 'ORDOS - SAY_POOL_OF_FIRE');
 DELETE FROM `creature_text` WHERE `entry`=72057 AND `groupid`=6 AND `id`=0;
 INSERT INTO `creature_text` (`entry`, `groupid`, `id`, `text`, `type`, `language`, `probability`, `emote`, `duration`, `sound`, `comment`) VALUES (72057, 6, 0, 'Burn!', 14, 0, 100, 0, 0, 38981, 'ORDOS - SAY_BURNING_SOUL');
+
+-- Database insert Error Fix
```
Comment (AriDEV) [inline sql/updates/world/creature_template_2018_03_09_00.sql]: phase mask is uint32 why changing the structure to uint8?

File: src/server/scripts/Pandaria/WorldBosses/boss_nalak.cpp [hunk 4]
```diff
@@ -69,19 +69,23 @@ class boss_nalak : public CreatureScript
...
+            introDone = false;
+        }
+
+        InstanceScript* pInstance;
+        bool introDone;
 
         EventMap events;
         SummonList summons;
 
-        void Reset() OVERRIDE
```
Comment (AriDEV) [inline src/server/scripts/Pandaria/WorldBosses/boss_nalak.cpp]: virtual functions need overrides

File: src/server/scripts/Pandaria/CMakeLists.txt [hunk 5]
```diff
@@ -11,20 +11,62 @@
...
+  Pandaria/Zones/TimelessIsle/Rares/rare_mob_rattleskew.cpp
+  Pandaria/Zones/TimelessIsle/Rares/rare_mob_rock_moss.cpp
+  Pandaria/Zones/TimelessIsle/Rares/rare_mob_spelurk.cpp
+  Pandaria/Zones/TimelessIsle/Rares/rare_mob_spirit_of_jadefire.cpp
+  Pandaria/Zones/TimelessIsle/Rares/rare_mob_stinkbraid.cpp
+  Pandaria/Zones/TimelessIsle/Rares/rare_mob_tsavo_ka.cpp
+  Pandaria/Zones/TimelessIsle/Rares/rare_mob_urdur_the_cauterizer.cpp
+  Pandaria/Zones/TimelessIsle/Rares/rare_mob_watcher_osu.cpp
+  Pandaria/Zones/TimelessIsle/Rares/rare_mob_zesqua.cpp
+  Pandaria/Zones/TimelessIsle/Rares/rare_mob_zhu_gon_the_sour.cpp
```
Comment (AriDEV) [inline src/server/scripts/Pandaria/CMakeLists.txt]: rares could be scripted in SAI, but if you want to script them in c++ its fine, but adding unfinished placeholder scripts is definitely the wrong idea script them first

File: sql/updates/world/creature_template_2018_03_09_00.sql (hunk 3 above)
Comment (Shoxxo) [reply sql/updates/world/creature_template_2018_03_09_00.sql]:
```sql
SQL(p): INSERT INTO creature (guid, id , map, spawnMask, phaseMask, modelid, equipment_id, position_x, position_y, position_z, orientation, spawntimesecs, spawndist, currentwaypoint, curhealth, curmana, MovementType, npcflag, unit_flags, dynamicflags) VALUES (8189273, 69099, 1064, 1, 4294967295, 0, 0, 7137.51, 5141.57, 99.7233, 2.35221, 300, 0, 0, 12725, 0, 0, 0, 0, 0)
 [ERROR]: [1264] Out of range value for column 'phaseMask' at row 1
Unhandled MySQL errno 1264. Unexpected behaviour possible.
````

There is your reason. I tested on this way and no more error. (Just a Temp Hackfix)

File: src/server/scripts/Pandaria/WorldBosses/boss_nalak.cpp (hunk 4 above)
Comment (Shoxxo) [reply src/server/scripts/Pandaria/WorldBosses/boss_nalak.cpp]: Sure, but if so or so, boss does nothing, but he can fly :)

File: src/server/scripts/Pandaria/CMakeLists.txt (hunk 5 above)
Comment (Shoxxo) [reply src/server/scripts/Pandaria/CMakeLists.txt]: The Tempholders should exist for references. i am not the only one here who spend time into skyfire as thanks i get asskicks and admin remove. so let other peoples do some stuff...

And SAI Yes ... 
screenshot_1

**Comment by Shoxxo** [comment]
So, what ever we do, the Boss Nalak and else Bosses with more complexe scripts will NOT work atm, i testes nalak in 5 script variants with some code changes and some tests. No way atm.

**Pull request closed without merging**